lib/Doctrine/ORM/EntityManager.php

```diff
@@ -156,6 +164,7 @@ protected function __construct(Connection $conn, Configuration $config, EventMan
         $this->metadataFactory->setEntityManager($this);
         $this->metadataFactory->setCacheDriver($this->config->getMetadataCacheImpl());
 
+        $this->persisterFactory  = new PersisterFactory($this);
```

Comment:
would it be possible to avoid creating yet another circular object graph ? Given that we consider that extending the EntityManager is not supported and that we encourage using composition instead, we should try to make it actually work. Anytime we pass ``$this`` to another object, we are making it impossible to use composition (because we pass the inner object, not the decorator).

Note that it might be very hard to do this in Doctrine 2.x (meaning that we are actually lying when saying they should decorate the EntityManager) because almost every low layers of the ORM are actually depending on the high level EntityManager (which itself depends on the lower layers)